### PR TITLE
[PACKAGE] Update kernel to 4.14.330

### PR DESCRIPTION
**Pull request by tiopex**
Use latest kernel 4.14.330 from https://www.kernel.org/ + miyoo patches

the updated kernel supports some new features e.g exFat

### CODE REVIEW

File: board/miyoo/kernel/0001-miyoo.patch
Comment (Apaczer) [inline board/miyoo/kernel/0001-miyoo.patch:1]: I like the idea, but maintaining/commiting kernel changes on a one big patch file will be a pain. Of course that's not an issue if you're working alone on a project

File: configs/miyoo_musl_defconfig [hunk 1]
```diff
@@ -22,9 +22,9 @@ BR2_ROOTFS_OVERLAY="board/miyoo/rootfs"
 BR2_ROOTFS_POST_IMAGE_SCRIPT="board/miyoo/scripts/genimage.sh"
 BR2_ROOTFS_POST_SCRIPT_ARGS="${BR2_TOOLCHAIN_BUILDROOT_VENDOR} ${BR2_VERSION_FULL} ${BR2_TOOLCHAIN_BUILDROOT_LIBC}"
 BR2_LINUX_KERNEL=y
-BR2_LINUX_KERNEL_CUSTOM_GIT=y
-BR2_LINUX_KERNEL_CUSTOM_REPO_URL="https://github.com/MiyooCFW/kernel.git"
-BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="origin/master"
+BR2_LINUX_KERNEL_CUSTOM_TARBALL=y
```
Comment (Apaczer) [inline configs/miyoo_musl_defconfig:25]: I understand you want to speed up build speed, but I would rather see grabing tarbal from cdn.kernel & then maybe grabing patch from our miyoocfw/kernel repo to apply.

Why won't you just rebase from 4.14.330 on our kernel and then add another diff commit on top with miyoo changes - for the sake of time. Might be wise to do it in PR to see what's happend. Lastly I would archive old kernel on another branch.

**Review by Apaczer — CHANGES_REQUESTED** [review]
:+1: Should we switch next to exFAT ? In future prospect I believe there could be [kernel 5.4](https://github.com/aodzip/buildroot-tiny200/blob/21de5725e119bece421236635a7815fe53aad532/board/widora/mangopi/r3/widora_mangopi_r3_defconfig#L45) support (aodzip have added even CedarX driver in his tree)

**Comment by tiopex** [comment]
> 👍 Should we switch next to exFAT ? In future prospect I believe there could be [kernel 5.4](https://github.com/aodzip/buildroot-tiny200/blob/21de5725e119bece421236635a7815fe53aad532/board/widora/mangopi/r3/widora_mangopi_r3_defconfig#L45) support (aodzip have added even CedarX driver in his tree)

Yes, switch to exfat and mount with sync 0, which should allow to switch off handheld without safe shutdown.

I have also working 4.19.x, tried also 5.4 but not booting, video drivers not working, audio driver cause panic and reset loop. Only backlight drivers works, the rest is unknown, so migrating to 5.4 will be problematic

Update: kernel 5.4 boot, however, can access only via USB serial, video and sound drivers not working. Video drivers load correctly with no any error in dmesg, audio driver causes crash and reset device

File: configs/miyoo_musl_defconfig (hunk 1 above)
Comment (tiopex) [reply configs/miyoo_musl_defconfig:25]: Ok, I will rebase our kernel to 4.14.330

**Comment by Apaczer** [comment]
What hold you to migrate to 4.19.x ?

**Comment by tiopex** [comment]
> What hold you to migrate to 4.19.x ?

Video is not stable. Sometimes not starting correctly

**Comment by Apaczer** [comment]
https://github.com/torvalds/linux/commit/b9d1e2e6265f5dc25e9f5dbfbde3e53d8a4958ac

Looks like exFAT has been added starting from 5.7
 I like the idea of sync 0 (was thinking of some periodic sync from daemon or main otherwise), maybe we could patch those changes in our 4.14?

**Comment by tiopex** [comment]
> [torvalds/linux@b9d1e2e](https://github.com/torvalds/linux/commit/b9d1e2e6265f5dc25e9f5dbfbde3e53d8a4958ac)
> 
> Looks like exFAT has been added starting from 5.7 I like the idea of sync 0 (was thinking of some periodic sync from daemon or main otherwise), maybe we could patch those changes in our 4.14?

weird https://www.kernelconfig.io/config_exfat_fs

anyway, I've managed kernel 5.4.x working which have exfat in staging

**Pull request closed without merging**